compiler/rustc_codegen_llvm/src/coverageinfo/mapgen/covfun.rs

```diff
@@ -128,15 +131,20 @@ fn fill_region_tables<'tcx>(
     let ffi::Regions { code_regions, branch_regions, mcdc_branch_regions, mcdc_decision_regions } =
         &mut covfun.regions;
 
+    let make_cov_span = |span: Span| {
+        spans::make_coverage_span(local_file_id, source_map, fn_cov_info, &source_file, span)
+    };
+
     // For each counter/region pair in this function+file, convert it to a
     // form suitable for FFI.
     let is_zero_term = |term| !covfun.is_used || ids_info.is_zero_term(term);
-    for Mapping { kind, ref source_region } in &fn_cov_info.mappings {
+    for &Mapping { ref kind, span } in &fn_cov_info.mappings {
         // If the mapping refers to counters/expressions that were removed by
         // MIR opts, replace those occurrences with zero.
         let kind = kind.map_terms(|term| if is_zero_term(term) { CovTerm::Zero } else { term });
 
-        let cov_span = ffi::CoverageSpan::from_source_region(local_file_id, source_region);
+        let Some(cov_span) = make_cov_span(span) else { continue };
```

Comment:
Question: with this `continue`, is it possible that we end up with zero available `cov_span`s in total?

Reply:
Yes, in rare cases this is possible. I believe that was the cause of the previous bug, because at that time the rest of the code wasn't prepared to deal with that possibility.

Reply:
Actually, it occurs to me that even if we don't have an “organic” way to reproduce the bug from source code, I could still rig up a perma-unstable flag that just tells this code to *always* discard every span, and have one test that enables that flag.

Reply:
That seems reasonable to me, yeah. I think it would be beneficial to have such a flag for testing purposes.

Reply:
There's already a `-Zcoverage-options` multi-purpose unstable flag, so I'm just going to add it as another value of that flag.

Reply:
That seems reasonable to me.